src/sas/qtgui/MainWindow/DataExplorer.py

```diff
@@ -519,7 +519,8 @@ def readProject(self, filename):
                 visible_perspective = value
             # send newly created items to the perspective
             self.updatePerspectiveWithProperties(key, value)
-
+        constraint_dict = self.readConstraintsFromProject(all_data)
+        self._perspective().updateFromConstraints(constraint_dict)
```

Comment:
`updateFromConstraints` is not available for other perspectives. Please move these 2 lines it a few lines below, after the explicit change to Fitting perspective.